include/drivers/bluetooth/hci_driver.h

```diff
@@ -95,6 +96,18 @@ int bt_recv(struct net_buf *buf);
  */
 int bt_recv_prio(struct net_buf *buf);
 
+/** @brief Read static addresses from the controller.
+ *
+ *  @note A weak version of this function is included in the HCI driver, so
+ *       defining it is optional per board.
```

Comment:
I guess you need to specify which HCI driver you're talking about, since I don't think you're implementing it for all HCI drivers. Also, mixing HCI driver and board in the same sentence is a bit confusing since there's no 1:1 relationship between those.

Reply:
Right. This comment out of sync now because I had to change my first approach, there is currently not a weak version of this function.
I think we should have one though. Since hci_core.c is the module that depends on this function I suggest to put the weak definition in there. And then rephrase this note:
`A weak version of this function is included so defining this is optional for the HCI driver`

So implementing this would be optional for someone writing an HCI driver.

Does that sound ok?

There is also a level below with `hci_vendor_read_static_addr` function in controller/hci/hci.c with weak version.
This is per vendor for the split LL. So that each vendor can define their own versions.

Reply:
Removed the comment.